src/NLog/Conditions/ConditionMethods.cs

```diff
@@ -166,5 +167,43 @@ public static int Length(string text)
         {
             return text.Length;
         }
+
+        /// <summary>
+        /// Indicates whether the specified regular expression finds a match in the specified input string.
+        /// </summary>
+        /// <param name="input">The string to search for a match.</param>
+        /// <param name="pattern">The regular expression pattern to match.</param>
+        /// <param name="options">A string consisting of the desired options for the test. Available options are : 'i' for ignore case, 'm' for multi-line, 's' for single line and 'x' for ignore pattern whitespace.</param>
+        /// <returns>true if the regular expression finds a match; otherwise, false.</returns>
+        [ConditionMethod("regex-matches")]
+        public static bool RegexMatches(string input, string pattern, string options)
+        {
+            RegexOptions regexOpts =
+#if SILVERLIGHT
+                RegexOptions.None;
+#else
+                RegexOptions.Compiled;
```

Comment:
please don't compile. never (well almost never). It could lead to memory issues, and the cached version is in 99% of the cases a better idea. 

The static methodes on Regex are already cached

Some articles about it: http://www.allben.net/post/2009/08/06/Performance-Compiled-vs-Interpreted-Regular-Expressions, https://blog.codinghorror.com/to-compile-or-not-to-compile/

Reply:
Since the regex specified will probably be used a lot and the doc says that you'll gain run-time performance by compiling, it made sense to me. I'll make it "not compiled" because it's only intuition.

Reply:
another option is to make it optional (off by default)